core/lib/zksync_core/src/consensus/storage/mod.rs

```diff
@@ -386,23 +436,18 @@ impl PayloadManager for Store {
         ctx: &ctx::Ctx,
         block_number: validator::BlockNumber,
     ) -> ctx::Result<validator::Payload> {
-        const POLL_INTERVAL: time::Duration = time::Duration::milliseconds(50);
-        loop {
-            let mut conn = self.access(ctx).await.wrap("access()")?;
-            if let Some(payload) = conn.payload(ctx, block_number).await.wrap("payload()")? {
-                let encoded_payload = payload.encode();
-                if encoded_payload.0.len() > 1 << 20 {
-                    tracing::warn!(
-                        "large payload ({}B) with {} transactions",
-                        encoded_payload.0.len(),
-                        payload.transactions.len()
-                    );
-                }
-                return Ok(encoded_payload);
-            }
-            drop(conn);
-            ctx.sleep(POLL_INTERVAL).await?;
+        tracing::info!("proposing block {block_number}");
+        let payload = self.wait_for_payload(ctx, block_number).await?;
+        let encoded_payload = payload.encode();
+        if encoded_payload.0.len() > 1 << 20 {
```

Comment:
Off-topic: The threshold should be extracted to a constant.

Reply:
done